### PR TITLE
[GWC-1171] Improve handling special characters in the GWC Demos Page

### PR DESCRIPTION
**Pull request by sikeoka**
https://github.com/GeoWebCache/geowebcache/issues/1171
This PR updates the GWC Demos Page to properly escape special characters from GeoServer catalog information.  One thing to note here is that StringEscapeUtils::escapeEcmaScript is not secure when used in HTML attributes so the OWASP Encoder library is used instead.

### CODE REVIEW

**Review by aaime — CHANGES_REQUESTED** [review]
Looks overall good, one QA failure.

File: geowebcache/core/src/main/java/org/geowebcache/demo/Demo.java [hunk 1]
```diff
@@ -45,11 +47,11 @@
 import org.geowebcache.mime.MimeType;
 import org.geowebcache.mime.XMLMime;
 import org.geowebcache.util.ServletUtils;
+import org.owasp.encoder.Encode;
 import org.springframework.util.Assert;
 
 public class Demo {
 
-    @SuppressWarnings("PMD.AvoidPrintStackTrace")
```
[the remark this answers is not in the document]
Comment (sikeoka) [reply geowebcache/core/src/main/java/org/geowebcache/demo/Demo.java:52]: Fixed.

**Comment by sikeoka** [comment]
There was a Java heap space problem.

**Review by aaime — APPROVED** [review]

**Comment by aaime** [comment]
The backport to `1.23.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 49b26ff7... [GWC-1171] Improve handling special characters in the GWC Demos Page
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging geowebcache/core/src/main/java/org/geowebcache/demo/Demo.java
CONFLICT (content): Merge conflict in geowebcache/core/src/main/java/org/geowebcache/demo/Demo.java

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-1.23.x 1.23.x
# Navigate to the new working tree
cd .worktrees/backport-1.23.x
# Create a new branch
git switch --create backport-1173-to-1.23.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 49b26ff7b0b49113d1e0ed3f45ffc52b094934eb
# Push it to GitHub
git push --set-upstream origin backport-1173-to-1.23.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-1.23.x
```
Then, create a pull request where the `base` branch is `1.23.x` and the `compare`/`head` branch is `backport-1173-to-1.23.x`.

**Pull request merged**